Remove NSFV Action and NSFV component, use Physics

GiudGiud opened the pull request:
refs #25642

GiudGiud committed:
Use Physics in lieu of action
refs #25642

GiudGiud committed:
Fix up retrieving the turbulence physics
refs #25642

GiudGiud committed:
More verbose assert for map sizes
More verbose block restriction error
refs #25642

GiudGiud committed:
Handle mixed types of inlet/outlet boundary conditions again
check the inputs before forming a map
refs #25642

GiudGiud committed:
Remove NSFVAction : covered by WCNSFV Physics
Remove FlowComponentNS: covered by FileMeshPhysicsComponent
refs #25642

GiudGiud committed:
Use FileMeshPhysicsComponent for flow components
Had to regold because using a single physics for both components and the two components were using different advection methods. This is not one of the block restricted parameters.
The other solution would have been to use two physics. This would necessitate different variable names at the moment. There were some work on merging physics together
It is possible, it's also quite a bit of code to achieve something that so far is not really needed
refs #25642

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2288314/) on cd222c6 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/27951/site)

This comment will be updated on new commits.

lindsayad commented:
Are the TH area users ready for this?

GiudGiud commented:
@licharlot @tanoret

GiudGiud commented:
Note that the old syntax remains completely functional. It's just the backend that changes (towards something that uses mostly the same code, just organized differently).
You do get a deprecation message.

GiudGiud commented:
Public app failure is fake

lindsayad commented:
> Note that the old syntax remains completely functional. It's just the backend that changes (towards something that uses mostly the same code, just organized differently).
> You do get a deprecation message.

Oh nice 👍

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2293194/) on c783c89 wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

### Navier stokes

<table>
  <thead>
    <tr>
      <th></th>
      <th>ed7323</th>
      <th colspan=4>#27951 c783c8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>84.00%</td>
      <td>84.57%</td>
      <td>+0.57%</td>
      <td>99.28%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>17323</td>
      <td>16183</td>
      <td>-1140</td>
      <td>276</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>3300</td>
      <td>2952</td>
      <td>-348</td>
      <td>2</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/27951/coverage_diff/navier_stokes)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/27951/coverage/navier_stokes)

### Thermal hydraulics

<table>
  <thead>
    <tr>
      <th></th>
      <th>ed7323</th>
      <th colspan=4>#27951 c783c8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>89.14%</td>
      <td>88.86%</td>
      <td>-0.28%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>14191</td>
      <td>14111</td>
      <td>-80</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1729</td>
      <td>1769</td>
      <td>+40</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/27951/coverage_diff/thermal_hydraulics)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/27951/coverage/thermal_hydraulics)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/stochastic_tools)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/27951/coverage/xfem)

</details>



This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2293194/) on c783c89 wanted to post the following:


The following coverage requirement(s) failed:
- `thermal_hydraulics` coverage rate 88.86% is less than the required 89.0%

joshuahansel commented on modules/navier_stokes/test/tests/finite_volume/ins/boussinesq/tests:
```diff
@@ -46,11 +46,13 @@
     requirement = 'The system should throw an error if the density is not a constant functor in case of Boussinesq treatment.'
     valgrind = 'none'
     issues = '#19742'
+    # New physics cannot support this because we want properties forwarded from fluid properties to work
```
I don't quite understand. So the Physics version is not going to take density as a functor?

joshuahansel replied:
So how does Physics affect this? And does this test make sense to exist? Do we just need to improve `isConstant` somehow?

GiudGiud replied:
>  Do we just need to improve isConstant somehow?

I dont see an easy way. Save for maybe evaluating it all over the mesh to check.

> So how does Physics affect this?

I dont want to limit incompressible flow to not using the FP properties and the GFFP. I ll give it another try to see if I can re-enable this.

joshuahansel reviewed (commented):
I want to discuss

> Had to regold because using a single physics for both components and the two components were using different advection methods. This is not one of the block restricted parameters.
The other solution would have been to use two physics. This would necessitate different variable names at the moment. There were some work on merging physics together
It is possible, it's also quite a bit of code to achieve something that so far is not really needed

In particular the part about needing different variables if needing more than one physics. I just want to understand any limitations before tighter integration between components and physics. But the change to this test is probably fine.

GiudGiud commented:
Unless we start changing the block restrictions of created objects, I don't see how we could not use different variable names. 

The physics create block restricted variables and UOs

joshuahansel commented:
Ok, I'm good with these changes, so just checking for any dissent by @licharlot or @tanoret before approval.

GiudGiud committed:
Add a transition tutorial and use snippets from real inputs
refs #25642

GiudGiud committed:
Set effective thermal conductivity to false in NSFVactions
The new default is true, as we have to assume the user does not want an extra multiplication by
porosity, as most PGH models do not require it
refs #25642

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2301559/) on 805ae63 wanted to post the following:


The following coverage requirement(s) failed:
- `thermal_hydraulics` coverage rate 88.86% is less than the required 89.0%

GiudGiud commented:
let's go?

licharlot commented:
Looks good to me. Thanks for adding the transition tutorial.

joshuahansel commented on modules/navier_stokes/src/actions/NSFVActionDeprecation.C:
```diff
@@ -25,7 +25,9 @@ NSFVActionDeprecation::NSFVActionDeprecation(const InputParameters & parameters)
   mooseDeprecated(
       "The NSFVAction, used through the [Modules/NavierStokesFV] syntax is deprecated. "
-      "Please convert your input file to using the [Physics/NavierStokes/WCNSFV...] syntax.");
+      "Please convert your input file to using the [Physics/NavierStokes/WCNSFV...] syntax.\n"
```
```suggestion
      "Please modify your input file to use the [Physics/NavierStokes/*] syntax.\n"
```

joshuahansel replied:
I see much or all of the syntax doesn't start with "WCNSFV"

joshuahansel commented on modules/navier_stokes/doc/content/syntax/Modules/NavierStokesFV/index.md:
```diff
@@ -149,6 +152,50 @@ energy source function can also be supplied to the incorporated
+notably because `/Modules/NavierStokes` has been changed to create the relevant `Physics` under the hood!
+
+To transition, you will have to split the `/Modules/NavierStokes` parameters into four groups below.
+Your simulation may only feature only one of these groups, in which case you will only need to create a single `Physics`.
```
```suggestion
Your simulation may only feature only one of these groups, in which case you will only need to create a single `Physics`:
```

joshuahansel commented on modules/navier_stokes/doc/content/syntax/Modules/NavierStokesFV/index.md:
```diff
@@ -149,6 +152,50 @@ energy source function can also be supplied to the incorporated
+!listing finite_volume/ins/channel-flow/2d-mixing-length-physics.i start=Physics/NavierStokes end=[Executioner]
+
+!alert note
+All `NavierStokes` `Physics` and be nested under the same `[NavierStokes]` sub-block. The separation in the examples into different `Physics` blocks is only for the purpose of simplifying the examples.
```
```suggestion
All `NavierStokes` `Physics` may be nested under the same `[NavierStokes]` sub-block. The separation in the examples into different `Physics` blocks is only for the purpose of simplifying the examples.
```

joshuahansel reviewed (changes requested):
Minor changes. For the transition doc, I think it would be useful to have 2 columns of code listing for the example - the before and after conversion for your example. And you could keep all of the physics in the same block and just discuss all afterward. But this is an optional change.

GiudGiud committed:
Address Josh's review

better language
more accurate help message

Co-authored-by: joshuahansel <joshua.hansel@inl.gov>

GiudGiud commented:
Can we merge this? 
If this triggers downstream VTB failure I'd like to be able to address them over the 4th

joshuahansel reviewed (approved):
Looks good. Sorry for the wait; I didn't notice you were ready for review again.

GiudGiud commented:
Thanks for the review! One last PR before being done with what was reported on the M2
